superset/db_engine_specs.py

```diff
@@ -1812,10 +1812,23 @@ def latest_sub_partition(cls, table_name, schema, database, **kwargs):
 
 class HiveEngineSpec(PrestoEngineSpec):
 
-    """Reuses PrestoEngineSpec functionality."""
+    """Based on PrestoEngineSpec but using hive date functions instead"""
 
     engine = 'hive'
     max_column_name_length = 767
+    time_grain_functions = {
+        None: '{col}',
+        'PT1S': "from_unixtime(unix_timestamp({col}), 'yyyy-MM-dd HH:mm:ss')",
```

Comment:
@jeepxiaozi I don't believe this logic is correct, i.e., it's assuming that the `{col}` is of type `STRING` where it could be a `DATE` or `TIMESTAMP`. I plan to address this issue in [SIP-15A](https://github.com/apache/incubator-superset/issues/7656) but for the meantime I would suggest implying that `{col}` can be successfully cast to a `TIMESTAMP`, i.e., `CAST({col} AS TIMESTAMP}`  (see [here](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+Types#LanguageManualTypes-TimestampstimestampTimestamps) for details) as this is consistent with the logic in the other engines. 

I'll address the invalid casting in SIP-15A which I'm actively working on.